### PR TITLE
Dotty partial support V2

### PR DESCRIPTION
**Pull request by kpbochenek**
Additional support for:
- export clauses
- given imports
- end markers
- super trait
- transparent modifier

### CODE REVIEW

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EndMarkerSuite.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    val code = """|object a { }
+                  |end a
+                  |
+                  |type K = Map
+                  |""".stripMargin
+    runTestAssert[Source](code, assertLayout = None)(
+      Source(
+        List(
+          Defn.Object(Nil, Term.Name("a"), Template(Nil, Nil, Self(Name(""), None), Nil)),
+          Term.EndMarker(Term.Name("a")),
```
Comment (tgodzik) [inline tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EndMarkerSuite.scala:38]: Should the end marker by a separate entity? Shouldn't we maybe have `TemplateWithEnd` and `BlockWithEnd` ?

Or alternatively, we could just ignore it in the parser? Especially I imagine when doing some scalafix rules or scalafmt formatting we will not want to have to search for the end token.

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 2]
```diff
@@ -178,6 +179,11 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       if (next.is[Whitespace] || next.is[Comment]) next.next
       else next
     }
+    def strictNext: Token = {
+      val next = scannerTokens.apply(Math.min(token.index + 1, scannerTokens.length - 1))
+      if (next.is[Space] || next.is[Tab] || next.is[Comment]) next.next
```
[the remark this answers is not in the document]
Comment (kpbochenek) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:184]: Good catch! next/prev do recursive call, I will do the same here to be consistent.
(will change `next.next` to `next.strictNext`)

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EndMarkerSuite.scala (hunk 1 above)
Comment (kpbochenek) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EndMarkerSuite.scala:38]: As discussed offline for now let's keep it separated and get back to it when we have significant indentation implemented and if `TemplateWithEnd` / `BlockWithEnd` are a way to go.

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala]: Did you change it?

[the remark this answers is not in the document]
Comment (kpbochenek) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala]: yes :+1:

**Pull request merged**